spec/lib/gakubuchi/template_spec.rb

```diff
@@ -1,24 +1,45 @@
 require 'rails_helper'
 
 RSpec.describe Gakubuchi::Template do
+  let(:template) { described_class.new(source_path) }
   let(:template_root) { described_class.root }
-  let(:template) { described_class.new(path) }
 
   describe '.all' do
     subject { described_class.all }
 
     let(:expectation) do
       [
-        described_class.new(described_class.root.join('foo.html.erb')),
-        described_class.new(described_class.root.join('bar/baz.html.erb')),
-        described_class.new(described_class.root.join('qux.html.haml')),
-        described_class.new(described_class.root.join('quux.html.slim'))
+        described_class.new('foo.html.erb'),
+        described_class.new('bar/baz.html.erb'),
+        described_class.new('qux.html.haml'),
+        described_class.new('quux.html.slim')
       ]
     end
 
     it { is_expected.to match_array(expectation) }
   end
 
+  describe '.new' do
+    subject { -> { described_class.new(source_path) } }
+
+    context 'when source path is relative' do
+      let(:source_path) { 'foo.html' }
```

Comment:
[IMO]
`foo.html.erb` seems better because in most cases the format of a source file is something other than `html` .